build/Autofac.Build.psm1

```diff
@@ -77,15 +69,39 @@ function Install-DotNetCli {
         }
 
         & ./.dotnet/dotnet-install.ps1 -InstallDir "$installDir" -Version $Version
-        $env:PATH = "$installDir;$env:PATH"
     } else {
         if (!(Test-Path ./.dotnet/dotnet-install.sh)) {
             Invoke-WebRequest "https://dot.net/v1/dotnet-install.sh" -OutFile "./.dotnet/dotnet-install.sh"
         }
 
         & bash ./.dotnet/dotnet-install.sh --install-dir "$installDir" --version $Version
-        $env:PATH = "$installDir`:$env:PATH"
     }
+
+    Extend-Path "$installDir"
+}
+
+<#
+.SYNOPSIS
+    Appends the Path with the given value but only if the value does not yet exist within the path.
+#>
+function Extend-Path {
+    [CmdletBinding()]
+    Param(
+        [string]
+        $path
+    )
+
+    if ($path -eq "") {
+      return;
+    }
+
+    $pathValues = $env:PATH.Split(";");
+
+    if ($pathValues -Contains $path) {
+      return;
+    }
+
+  $env:PATH = "$path;$env:PATH"

```

Comment:
```suggestion
<#
.SYNOPSIS
    Appends a given value to the path but only if the value does not yet exist within the path.
.PARAMETER Path
    The path to append.
#>
function Add-Path {
    [CmdletBinding()]
    Param(
        [ValidateNotNullOrEmpty()]
        [string]
        $Path
    )

    $pathValues = $env:PATH.Split(";");

    if ($pathValues -Contains $Path) {
      return;
    }

  $env:PATH = "$Path;$env:PATH"
```

- Added parameter doc
- Added validation by attribute
- Renamed to use [approved PowerShell verb](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.1&viewFallbackFrom=powershell-6) `Add` for consistency

Reply:
Thanks, applied.